spikeextractors/extractors/nwbextractors/nwbextractors.py

```diff
@@ -568,22 +569,32 @@ def add_electrodes(recording: se.RecordingExtractor, nwbfile=None, metadata: dic
 
                 if not any([x.get('name', '') == 'group_name' for x in metadata['Ecephys']['Electrodes']]):
                     group_id = recording.get_channel_groups(channel_ids=channel_id)[0]
-                    if group_id in range(len(nwbfile.electrode_groups)):
-                        group_name = list(nwbfile.electrode_groups.keys())[group_id]
+                    if str(group_id) in nwbfile.electrode_groups:
                         electrode_kwargs.update(
                             dict(
-                                group=nwbfile.electrode_groups[group_name],
-                                group_name=group_name
+                                group=nwbfile.electrode_groups[str(group_id)],
+                                group_name=str(group_id)
                             )
                         )
                     else:
-                        warnings.warn("No metadata was passed specifying the electrode group for "
-                                      f"electrode {channel_id}, and the internal recording channel group was "
-                                      f"assigned a value ({group_id}) outside the indices of the electrode "
-                                      "groups in the nwbfile! Electrode will not be added.")
-                        continue
+                        if group_id in range(len(nwbfile.electrode_groups)):
```

Comment:
Why leaving this condition? 

If channel groups are present and the `NwbRecordingExtractor` is used to write to nwb, it will always go into the previous if (`if str(group_id) in nwbfile.electrode_groups`). In case it gets here, I think we just need to raise a warning. using the `range` might in fact select the wrong electrode group anyways. 

@CodyCBakerPhD what do you think?

Reply:
Removed condition. Though it may prevent the function from writing electrodes in particular cases (passing metadata for groups but not electrodes), this will ensure the information of the linking is more accurate and makes fewer assumptions.